### PR TITLE
ODSC-39737 : allow to use `GenericModel.predict()` locally

### PR DESCRIPTION
**Pull request by mingkang111**
# Description
Related JIRA:https://jira.oci.oraclecorp.com/browse/ODSC-39737
Currently, the GenericModel supports two methods that can be used to test the model:
* verify()
* predict()

The difference between these two methods is that verify is used for local testing, where predict sends the request to the ML endpoint. Having two different methods confuses users, they don't really understand why we need two different methods for inferring the model. 

# Changed 
* Add a new attribute - `local=True/False` to the predict method to be able to invoke the prediction locally.

# User Experience
```
prediction = GenericModel.predict(local=True)
```
Screenshot 2023-04-05 at 3 54 17 PM

# Test
Screenshot 2023-03-29 at 1 08 10 PM

## Int test in TC
TC: https://teamcity.oci.oraclecorp.com/buildConfiguration/Odsc_Ads_AdsCiCd_CiTrigger/38127161?buildTab=buildParameters
Screenshot 2023-04-05 at 3 51 59 PM

### CODE REVIEW

**Review by qiuosier — APPROVED** [review]

File: tests/unitary/with_extras/model/test_generic_model.py [hunk 1]
```diff
@@ -168,7 +168,20 @@
     "training_script": None,
 }
 
-
+INFERENCE_CONDA_ENV= "oci://service-conda-packs@ociodscdev/service_pack/cpu/General_Machine_Learning_for_CPUs/1.0/mlcpuv1"
```
Comment (mrDzurb) [inline tests/unitary/with_extras/model/test_generic_model.py:171]: We don't need to use real paths, right?

**Review by mrDzurb — APPROVED** [review]

File: tests/unitary/with_extras/model/test_generic_model.py (hunk 1 above)
Comment (mingkang111) [reply tests/unitary/with_extras/model/test_generic_model.py:171]: Yes, we only need slug name I think. I will create a PR to clean up the paths in this file.

**Review by mayoor — DISMISSED** [review]
As a followup, I have a proposal - 

If user calls `predict` and the `url` is None, when we throw an error, we can provide a message saying, `Error invoking the remote endpoint as the model is not deployed yet or the endpoint information is not available. If you intend to invoke inference using locally available model artifact, set parameter local=True`

**Review by liudmylaru — DISMISSED** [review]

**Comment by mingkang111** [comment]
> As a followup, I have a proposal -
> 
> If user calls `predict` and the `url` is None, when we throw an error, we can provide a message saying, `Error invoking the remote endpoint as the model is not deployed yet or the endpoint information is not available. If you intend to invoke inference using locally available model artifact, set parameter local=True`

File: tests/unitary/with_extras/model/test_generic_model.py (hunk 1 above)
Comment (mingkang111) [reply tests/unitary/with_extras/model/test_generic_model.py]: Replaced with fake path.

**Comment by mingkang111** [comment]
> As a followup, I have a proposal -
> 
> If user calls `predict` and the `url` is None, when we throw an error, we can provide a message saying, `Error invoking the remote endpoint as the model is not deployed yet or the endpoint information is not available. If you intend to invoke inference using locally available model artifact, set parameter local=True`

That's a good idea! I just improved the error message as suggested. Thanks Mayoor.

**Review by mrDzurb — APPROVED** [review]

**Review by mayoor — APPROVED** [review]

**Pull request merged**